src/plugins/intel_npu/src/plugin/npuw/compiled_model.cpp

```diff
@@ -406,24 +412,51 @@ ov::npuw::CompiledModel::CompiledModel(const std::shared_ptr<ov::Model>& model,
     reset_io();
 }
 
-void ov::npuw::CompiledModel::fill_weights_bank(const std::size_t idx) {
-    LOG_VERB("Filling weights bank for Subgraph[" << idx << "]...");
-    LOG_BLOCK();
+void ov::npuw::CompiledModel::finalize_weights_bank() {
+    // Evaluate lazy tensors which aren't in the bank beforehand
+    std::vector<std::vector<ov::Tensor>> evaluated_tensors(m_compiled_submodels.size());
```

Comment:
why should we store them all here? Can't we just
1. Walk over `CompiledModelDesc`s
2. Walk over their `_lazy_closure`s
3. Pass `_lazy_closure`s to the bank?

Bank decides if it is a hit or miss based on LazyClosure itself (as a key) - and evaluates and copies it to the device memory IF it was a miss.

Reply:
As I mentioned in the comment above it's already a couple of seconds even with `parallel_for`